printing an ast.Node causes bad C generation

```
OS: macos, macOS, 12.0.1, 21A559
Processor: 10 cpus, 64bit, little endian, Apple M1 Max
CC version: Apple clang version 13.0.0 (clang-1300.0.29.3)

getwd: /Users/thomas/Unix/v
vmodules: /Users/thomas/.vmodules
vroot: /Users/thomas/Unix/v
vexe: /Users/thomas/Unix/v/v
vexe mtime: 2021-12-02 09:32:01
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 5c8e626.5ab91dd
env VFLAGS: "-g -stats -d panics_break_into_debugger"

Git version: git version 2.34.0
Git vroot status: weekly.2021.47-84-g5ab91dd4
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 689c8a02
```

**What did you do?**

```diff
diff --git a/vlib/v/transformer/transformer.v b/vlib/v/transformer/transformer.v
index 5f7f0e7fa..b5d00dc26 100644
--- a/vlib/v/transformer/transformer.v
+++ b/vlib/v/transformer/transformer.v
@@ -37,6 +37,13 @@ pub fn (t Transformer) stmt(mut node ast.Stmt) {
                        for mut right in node.right {
                                right = t.expr(right)
                        }
+                       right := node.right[0]
+                       match right {
+                               ast.ArrayInit {
+                                       eprintln(right)
+                               }
+                               else {}
+                       }
                }
                ast.Block {
                        for mut stmt in node.stmts {
```

**What did you expect to see?**

No impact on the C generated.

**What did you see instead?**

Some compilation error.

Actually no, @spytheman's example works, but after adding the code block to transformer.v, I get

```
                                                                                         ^
/tmp/v_501/v2.10070068966803326283.tmp.c:17702:75: error: source file is not valid UTF-8
        Array_u8 res = __new_array_with_default(exp + 32, 0, sizeof(u8), &(u8[]){<C1>});
                                                                                 ^
/tmp/v_501/v2.10070068966803326283.tmp.c:21692:45: warning: null character ignored [-Wnull-character]
        Array_u8 res = __new_array_with_default(0, <U+0000>, sizeof(u8), 0);
                                                   ^
/tmp/v_501/v2.10070068966803326283.tmp.c:21692:46: error: expected expression
        Array_u8 res = __new_array_with_default(0, <U+0000>, sizeof(u8), 0);
                                                           ^
/tmp/v_501/v2.10070068966803326283.tmp.c:22320:52: warning: null character ignored [-Wnull-character]
        Array_RepIndex idxs = __new_array_with_default(0, <U+0000>, sizeof(RepIndex), 0);
```

--- a/vlib/v/ast/str.v
+++ b/vlib/v/ast/str.v
@@ -346,13 +346,13 @@ pub fn (x Expr) str() string {
 			return '.$x.val'
 		}
 		FloatLiteral, IntegerLiteral {
-			return x.val
+			return '$x.val'
 		}
 		GoExpr {
 			return 'go $x.call_expr'
 		}
 		Ident {
-			return x.name
+			return '$x.name'
 		}
 		IfExpr {
 			mut parts := []string{}

--- a/vlib/v/ast/str.v
+++ b/vlib/v/ast/str.v
@@ -346,13 +346,13 @@ pub fn (x Expr) str() string {
 			return '.$x.val'
 		}
 		FloatLiteral, IntegerLiteral {
-			return '$x.val'
+			return x.val.clone()
 		}
 		GoExpr {
 			return 'go $x.call_expr'
 		}
 		Ident {
-			return '$x.name'
+			return x.name.clone()
 		}
 		IfExpr {
 			mut parts := []string{}

